cpp/src/io/orc/reader_impl.cu

```diff
@@ -414,20 +450,26 @@ rmm::device_buffer reader::impl::decompress_stripe_data(
         break;
       default: CUDF_EXPECTS(false, "Unexpected decompression dispatch"); break;
     }
+    decompress_check(inflate_out.data(), num_compressed_blocks, stream, any_page_failure.device_ptr());
   }
   if (num_uncompressed_blocks > 0) {
     CUDA_TRY(gpu_copy_uncompressed_blocks(
       inflate_in.data() + num_compressed_blocks, num_uncompressed_blocks, stream));
   }
   gpu::PostDecompressionReassemble(compinfo.device_ptr(), compinfo.size(), stream);
 
+  any_page_failure.device_to_host(stream);
+
   // Update the stream information with the updated uncompressed info
   // TBD: We could update the value from the information we already
   // have in stream_info[], but using the gpu results also updates
   // max_uncompressed_size to the actual uncompressed size, or zero if
   // decompression failed.
```

Comment:
This comment block belongs above the for loop below.

Reply:
Fixed